internal/api/courses/assignments/submissions/proxy/resubmit.go

```diff
@@ -0,0 +1,76 @@
+package proxy
+
+import (
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/db"
+	"github.com/edulinq/autograder/internal/grader"
+	"github.com/edulinq/autograder/internal/timestamp"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+type ResubmitRequest struct {
+	core.APIRequestAssignmentContext
+	core.MinCourseRoleGrader
+	TargetSubmission string `json:"target-submission"`
+
+	ProxyUser core.TargetCourseUser `json:"proxy-email"`
+	ProxyTime *timestamp.Timestamp  `json:"proxy-time"`
+
+	Message string `json:"message"`
+}
+
+type ResubmitResponse struct {
+	core.BaseSubmitResponse
+
+	FoundUser       bool `json:"found-user"`
+	FoundSubmission bool `json:"found-submission"`
+}
+
+// Proxy resubmit an assignment submission to the autograder.
+func HandleResubmit(request *ResubmitRequest) (*ResubmitResponse, *core.APIError) {
+	response := ResubmitResponse{}
+
+	if !request.ProxyUser.Found {
+		return &response, nil
+	}
+
+	response.FoundUser = true
+
+	gradingResult, err := db.GetSubmissionContents(request.Assignment, request.ProxyUser.Email, request.TargetSubmission)
+	if err != nil {
+		return nil, core.NewInternalError("-632", &request.APIRequestCourseUserContext, "Failed to get submission contents.").
+			Err(err).Assignment(request.Assignment.GetID()).
```

Comment:
Just an FYI, there should be (but is not) an error function for core.APIRequestAssignmentContext.
I'm going to make one, so you will probably have to deal with a merge
(depending on when it gets in).